technique/muon/zero_field_setup.py

```diff
@@ -0,0 +1,269 @@
+        """
+        Read corrected values from the OPI
+        :return: corrected values for all axis
+        """
+        x = self.get_single_corrected_field_value("X")
+        y = self.get_single_corrected_field_value("Y")
+        z = self.get_single_corrected_field_value("Z")
+
+        return x, y, z
+
+    def get_magnitude(self):
+        """
+        Reads magnitude value from the OPI
+        :return: magnitude of magnetic field
+        """
+        return g.get_pv(PV_READ_MAGNETIC_FIELD_STRENGTH)
+
+    def set_fields_setpoint_to_zero(self):
+        """
+        Set all set points to 0
+        :param value_x: target value for x axis
+        :param value_y: target value for y axis
+        :param value_z: target value for z axis
+        :return: None
+        """
+        g.set_pv(PV_FIELD_SETPOINT.format("X"), 0, is_local=True)
+        g.set_pv(PV_FIELD_SETPOINT.format("Y"), 0, is_local=True)
+        g.set_pv(PV_FIELD_SETPOINT.format("Z"), 0, is_local=True)
+
+    def get_correlated_current_and_fields(self, plot=False):
+        """
+        Gets correlated current and corrected fields, 2 second wait before taking the field
+        reading in all three axis
+        :param plot: data to be plotted or not
+        :return: correlated current and fields for each axis
+        """
+        ps_lower_limit_x = g.get_pv(PV_POWER_SUPPLY_LOWER_LIMIT.format("X"))
+        ps_lower_limit_y = g.get_pv(PV_POWER_SUPPLY_LOWER_LIMIT.format("Y"))
+        ps_lower_limit_z = g.get_pv(PV_POWER_SUPPLY_LOWER_LIMIT.format("Z"))
+
+        ps_max_x = abs(g.get_pv(PV_POWER_SUPPLY_UPPER_LIMIT.format("X"), is_local=True) - \
+                   ps_lower_limit_x)
+
+        ps_max_y = abs(g.get_pv(PV_POWER_SUPPLY_UPPER_LIMIT.format("Y"), is_local=True) - \
+                   ps_lower_limit_y)
+
+        ps_max_z = abs(g.get_pv(PV_POWER_SUPPLY_UPPER_LIMIT.format("Z"), is_local=True) - \
+                   ps_lower_limit_z)
+
+        scale_x = ps_max_x / 20
+        scale_y = ps_max_y / 20
+        scale_z = ps_max_z / 20
+
+        current_x = []
+        current_y = []
+        current_z = []
+
+        fields_x = []
+        fields_y = []
+        fields_z = []
+
+        # 21 evenly spaced steps
+        for x in range(21):
+            current_x.append(ps_lower_limit_x + (x * scale_x))
+            current_y.append(ps_lower_limit_y + (x * scale_y))
+            current_z.append(ps_lower_limit_z + (x * scale_z))
+
+            g.set_pv(PV_POWER_SUPPLY_SET_POINT.format("X"), ps_lower_limit_x + (x * scale_x))
+            g.set_pv(PV_POWER_SUPPLY_SET_POINT.format("Y"), ps_lower_limit_y + (x * scale_y))
+            g.set_pv(PV_POWER_SUPPLY_SET_POINT.format("Z"), ps_lower_limit_z + (x * scale_z))
+
+            time.sleep(2)
+
+            fields_x.append(self.get_single_corrected_field_value("X"))
+            fields_y.append(self.get_single_corrected_field_value("Y"))
+            fields_z.append(self.get_single_corrected_field_value("Z"))
```

Comment:
I know the wiki page didn't explicitly state this (sorry), but it would be a good idea to put the power supplies back to the middle of their range rather than being right on a limit after the loop finishes. That will give us a more representative stability graph when we move on to plotting the stabilities.